File: test-suite-kotlin/src/main/kotlin/example/Feature.kt
package example

import io.micronaut.core.convert.ConversionContext
import io.micronaut.core.convert.TypeConverter
import jakarta.inject.Singleton
import java.util.*

class Feature(private val name: String) {
    fun name(): String {
        return name
    }

    override fun toString(): String { // <1>
        return name
    }
}

@Singleton
class FeatureConverter : TypeConverter<String, Feature> {
    // <2>
    override fun convert(value: String, targetType: Class<Feature>, context: ConversionContext): Optional<Feature> {
        return Optional.of(Feature(value))
    }
}

File: doc-examples/example-python/src/main/python/example/Feature.py
from jakarta.inject import Singleton
from java.util import Optional
from micronaut.core.convert import ConversionContext, TypeConverter


class Feature:
    def __init__(self, name: str):
        self._name = name

    def name(self) -> str:
        return self._name

    def __str__(self) -> str:  # <1>
        return self._name


@Singleton
class FeatureConverter(TypeConverter[str, Feature]):  # <2>
    def convert(self, object: str, target_type: type[Feature], context: ConversionContext):
        return Optional.of(Feature(object))
